bpf/bpf_lxc.c

```diff
@@ -1591,7 +1591,11 @@ int tail_ipv6_policy(struct __ctx_buff *ctx)
 	ctx_store_meta(ctx, CB_FROM_HOST, 0);
 
 #ifdef HAVE_ENCAP
-	from_tunnel = ctx_load_meta(ctx, CB_FROM_TUNNEL);
+	/* TODO use CB_FROM_TUNNEL on v1.16, when we can trust that all
+	 * callers populate it (even after downgrade).
+	 */
+	/* from_tunnel = ctx_load_meta(ctx, CB_FROM_TUNNEL); */
+	from_tunnel = true;
```

Comment:
Don't `CILIUM_CALL_IPV6_TO_LXC_POLICY_ONLY` get called from `cilium_host` for host -> local pod requests? If yes, then we are marking such a traffic as `from_tunnel`.

Reply:
yep, that's the old behaviour - we were classifying everything as (potential) tunnel traffic. Takes just a bit longer now until we can be precise, and limit the processing to actual tunnel traffic.